proto/minder/v1/minder.proto

```diff
@@ -2626,3 +2626,42 @@ message EvaluationHistoryAlert {
     // the structure and contents are alert specific, and are subject to change.
     string details = 2;
 }
+
+message SelectorEntity {
+    // one of repository, pull_request, artifact (see oneof entity)
+    Entity entity_type = 1;
+    // the name of the entity, same as the name in the entity message
+    string name = 2;
```

Comment:
I wanted to add the provider name as a separate attribute alongside name when we add fetching of the key-value pairs from the provider. Since you [suggested](https://github.com/stacklok/minder/pull/3797#discussion_r1669002640) to just use full_name were you thinking to make the provider name part of the full name as well?

I like how unambiguous that would be, but if we don't provide a very easy way "just ignore this repo" especially from the UI then the grammar gets less useful to the user.

Reply:
I think we might want one property for the `full_name`, and one property for the `provider`.  That would allow for full distinguishing, but would default to names that were recognizable to people without some sort of special prefix.

Reply:
It might also simplify "apply this policy to all `billing/fooserver` images at all registries under Minder automation"

Reply:
I added a provider message to the messages is that what you were describing? There's name and provider class exposed for start.

Reply:
**question (non-blocking):** is there a particular reason why we're not calculating an _exact_, possibly unambiguous name and storing it alongside the entity? It really feels like we should have for repositories something like a URI/purl alongside repo name, owner, and provider, something like `<provider>:<owner>:<name>`, e.g. `github-app-foo:stacklok:minder`. On top of that, we can define custom functions to ship with the CEL evaluator that operate on top of that.

Reply:
We can do this as well and Evan suggested something similar. Good idea, thanks.